modules/programs/terraform.nix

```
@@ -0,0 +1,55 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.terraform;
+
+in {
+  meta.maintainers = [ maintainers.bryanhonof ];
+
+  options.programs.terraform = {
+
+    enable = mkEnableOption ''
+      Terraform is an open-source infrastructure as code software tool that
+      provides a consistent CLI workflow to manage hundreds of cloud services.
+      Terraform codifies cloud APIs into declarative configuration files.
+
+      https://www.terraform.io/
+    '';
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.terraform;
+      defaultText = "pkgs.terraform";
+      description = "The Terraform package to use";
+    };
+
+    providers = mkOption {
+      type = types.listOf types.package;
+      default = [ ];
+      description = "A list of Terraform providers.";
+    };
+
+    installCompletion = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Wether or not to install the autocomplete functionality. Normally this
+        is done by running <literal>terraform -install-autocomplete</literal>.
+        Currently only installs for bash.
+      '';
+    };
```

The pattern we usually use is `enable${SHELL}Integration`.

```suggestion
    enableBashIntegration = mkEnableOption "Bash integration" // {
      default = true;
    };
```